src/helpers/curve.ts

```diff
@@ -119,19 +113,17 @@ export function getAmountAFromLiquidityDelta(
   maxSqrtPrice: BN,
   rounding: Rounding
 ): string {
-  const prod = new Decimal(liquidity.toString()).mul(
-    new Decimal(maxSqrtPrice.sub(currentSqrtPrice).toString())
-  );
+  // Q128.128
+  const product = liquidity.mul(maxSqrtPrice.sub(currentSqrtPrice));
+  // Q128.128
   const denominator = currentSqrtPrice.mul(maxSqrtPrice);
-  // prod: Q128.128, denominator: Q128.128
-  const result = prod
-    .mul(Decimal.pow(2, 64))
-    .div(new Decimal(denominator.toString()));
-
+  const one = new BN(1).shln(64);
+  // Q64.64
+  const result = product.shln(64).div(denominator);
   if (rounding == Rounding.Up) {
-    return result.div(Decimal.pow(2, 64)).ceil().toFixed();
+    return result.add(one).shrn(64).toString();
```

Comment:
that is wrong for rounding up calculation, pls refer how program doing now

Reply:
fixed